unity-renderer/Assets/UnityGLTF/Scripts/GLTFComponent.cs

```diff
@@ -388,34 +385,34 @@ private void OnDestroy()
             CleanUp();
 
             if (state != State.COMPLETED)
-            {
-                ctokenSource.Cancel();
-                ctokenSource.Dispose();
-            }
+                Cancel();
+        }
+
+        public void Cancel()
+        {
+            if (ctokenSource == null) return;
+            ctokenSource.Cancel();
+            ctokenSource?.Dispose();
+            ctokenSource = null;
         }
+
         private void CleanUp()
         {
             scheduledToCleanUp = true;
 
-            if (state == State.QUEUED)
-            {
-                DequeueDownload();
-            }
+            if (state == State.QUEUED) { DequeueDownload(); }
 
-            if (state == State.DOWNLOADING)
-            {
-                DecrementDownloadCount();
-            }
+            if (state == State.DOWNLOADING) { DecrementDownloadCount(); }
 
-            if (!alreadyLoadedAsset)
-            {
-                OnFail_Internal(null);
-            }
+            if (!alreadyLoadedAsset) { OnFail_Internal(null); }
 
             state = State.NONE;
         }
 
-        bool IDownloadQueueElement.ShouldPrioritizeDownload() { return prioritizeDownload; }
+        bool IDownloadQueueElement.ShouldPrioritizeDownload()
```

Comment:
As it is a property, this one should be single lined

`bool IDownloadQueueElement.ShouldPrioritizeDownload() => prioritizeDownload;`